### DISCUSSION TITLE
Issue with Filtering Using Logical Operators

### DISCUSSION THREAD

**Discussion by MainqueQwavee** [Adrinalin4ik/Nestjs-Graphql-Tools#42] [Q&A] [answered]
I'm experiencing issues with filtering using logical operators in our current implementation. Specifically, I'm trying to apply a filter where multiple conditions are combined using AND, and one of these conditions involves an OR between two fields. However, I haven't been able to get this working as expected.

Here's an example of what I'm trying to achieve:

    Condition 1: Field A equals value X
    Condition 2: (Field B equals value Y OR Field C equals value Z)
    Condition 3: Field D is greater than value W

Currently, the library provides _and_ and or _fields_. As I understand it, the _and_ field is an array of filters where each filter is combined using AND comparatives, while the _or_ field is another array of filters combined using OR comparatives. In my case, I need to use a combination of both.

However, the current implementation does not seem to support this structure, or there is a misunderstanding on how to properly use the logical operators in this context.

**Example Query with TypeORM QueryBuilder**

Let's assume we have an entity called _ExampleEntity_ with the following fields: _fieldA_, _fieldB_, _fieldC_, and _fieldD_.
Here’s an example of the query using QueryBuilder in TypeORM:

```
async function getFilteredResults() {
  const repository = getRepository(ExampleEntity);
  
  const results = await repository.createQueryBuilder('example')
    .where('example.fieldA = :valueA', { valueA: 'X' })
    .andWhere(
      new Brackets(qb => {
        qb.where('example.fieldB = :valueB', { valueB: 'Y' })
          .orWhere('example.fieldC = :valueC', { valueC: 'Z' });
      })
    )
    .andWhere('example.fieldD > :valueD', { valueD: 'W' })
    .getMany();

  return results;
}
```

**Questions:**

    Is the current implementation capable of handling logical operations like the one described above?
    If so, could you provide an example of the correct syntax or approach?
    If not, are there plans to support this functionality in the future?

Thank you for your assistance!

**Comment by Adrinalin4ik** [comment, marked as answer]
@MainqueQwavee Hey! Thanks for the feedback. Just implemented this feature in the 1.10.0 version https://github.com/Adrinalin4ik/Nestjs-Graphql-Tools/releases/tag/0.10.0

**Reply by MainqueQwavee** [reply]
Hi @Adrinalin4ik! Thank you so much for the quick update. I've tested the new feature in version 1.10.0, and it's working perfectly! I really appreciate your support and the quick turnaround.
